### PR TITLE
Tidy-ups

### PR DESCRIPTION
**Pull request by luke-gumbley**
This PR changes the publishing options to a single executable, dependent on runtime, and removed some redundant assembly references (release zip should now be <1mb).

I also resolved all outstanding warnings - sorry, that's why this PR is so large. No functionality changes.

### CODE REVIEW

File: AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs [hunk 1]
```diff
@@ -52,15 +52,15 @@ public struct keyboardHookStruct
...
-        public static extern IntPtr GetModuleHandle(string lpModuleName);
+        public static extern IntPtr GetModuleHandle(string? lpModuleName);
 
         #endregion
 
 
         public void Start()
         {
             HookProcessor = Hook_Callback;
-            IntPtr hInstance = GetModuleHandle(Process.GetCurrentProcess().MainModule.ModuleName);
```
Comment (uxsoft) [inline AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs:63]: Are you sure this still works?

File: AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs [hunk 2]
```diff
@@ -52,15 +52,15 @@ public struct keyboardHookStruct
...
+            IntPtr hInstance = GetModuleHandle(null);
```
Comment (uxsoft) [inline AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs:63]: Are you sure this still works?

**Comment by luke-gumbley** [comment]
Yep - passing null to GetModuleHandle returns a handle to the file used to create the calling process:
https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulehandlea

I've also tested that Hook_Callback still executes with this change in place.

File: AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs (hunk 2 above)
Comment (luke-gumbley) [reply AppleWirelessKeyboardCore/Keyboard/WHKeyboardLLModule.cs:63]: Yep - passing null to GetModuleHandle returns a handle to the file used to create the calling process:
https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulehandlea

I've also tested that Hook_Callback still executes with this change in place.

**Pull request merged**